lib/cext/preprocess.rb

```diff
@@ -30,6 +35,36 @@
 
       "#{shim}(#{argc}, #{argv}, \"#{arity}\", "
     }
+  elsif line =~ /^(\s+)VALUE((\s+\w+\s*(\[\s*\d+\s*\]\s*)?,?)+);\s*$/
+    # Translate
+    #   VALUE args[6], failed, a1, a2, a3, a4, a5, a6;
+    #  into
+    #   VALUE failed, a1, a2, a3, a4, a5, a6; VALUE *args = truffle_managed_malloc(6 * sizeof(VALUE));
+
+    simple = []
+    arrays = []
+
+    line = $1
+
+    $2.split(', ').each do |local|
+      local.strip!
+      if local.end_with?(']')
+        local =~ /(\w+)\s*\[\s*(\d+)\s*\]/
+        arrays.push [$1, $2.to_i]
```

Comment:
I think we're debating a very minor point of style here. If you think it's error prone or confusing then I'll listen to that argument. Otherwise it's just my opinion versus yours.